src/repository/favouriteRepository.js

```diff
@@ -0,0 +1,19 @@
+const favouriteRepository = {
+  findAll: () => {
+    if (typeof window.Storage !== 'undefined') {
+      const encodedFavourites = window.localStorage.getItem('favourites');
+      if (encodedFavourites !== null) {
+        return JSON.parse(encodedFavourites);
+      }
+    }
+
+    return [];
+  },
+  save: (favourites) => {
```

Comment:
Mógłbyś przerobić to api z `save` całości na `saveItem/deleteItem`.
Dzięki temu zenkapsulowałbyś w tym repozytorium logikę ogarniania kolekcji i komponenty mogłyby tylko wołać funkcje w zapisz/usuń, a nie trzebaby tego 'splice/push' wywoływać w komponencie tylko byłoby tutaj.
Ale to raczej opinia, tak jak jest też jest ok.